Modules/xSharePoint/DSCResources/MSFT_xSPOutgoingEmailSettings/MSFT_xSPOutgoingEmailSettings.psm1

```diff
@@ -0,0 +1,94 @@
+function Get-TargetResource
+{
+	[CmdletBinding()]
+	[OutputType([System.Collections.Hashtable])]
+	param
+	(
+		[parameter(Mandatory = $true)] [System.String] $WebAppUrl,
+		[parameter(Mandatory = $true)] [System.String] $SMTPServer,
+		[parameter(Mandatory = $true)] [System.String] $FromAddress,
+		[parameter(Mandatory = $true)] [System.String] $ReplyToAddress,
+		[parameter(Mandatory = $true)] [System.String] $CharacterSet
+		
+	)
+
+	Write-Verbose -Message "Retrieving outgoing email settings configuration "
+
+	$result = Invoke-xSharePointCommand -Arguments $PSBoundParameters -ScriptBlock {
+		$params = $args[0]
+		
+		$webApp = $null
+		if([string]::IsNullOrEmpty($params.WebAppUrl)){
```

Comment:
If WebAppUrl is the key value for this resource, what's the scenario where it can be null or empty? I'll test it in my lab but I'm not sure you'll be able to pass an empty WebAppUrl value in a configuration